spec/integration/filtering_spec.rb

```diff
@@ -66,6 +66,47 @@ def run_rerun_command_for_failing_spec
   end
 
   context "passing a line-number filter" do
+    it "works with different custom runners used in the same process" do
+      class Foo
```

Comment:
I'd give this a better name, and make this an anonymous class, something like:
```suggestion
      result_counter = Class.new do
        RSpec::Core::Formatters.register(self, :example_passed)
```